### PR TITLE
modernize `TkAlCaRecoMonitor` source and configuration

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Title says it all:
   * add a `fillDescriptions`;
   * use range-based loops;
   * reduce code duplication. 

#### PR validation:

   * `cmssw` compiles;
   * `scram b runtests_testDiMuonVertexMonitor use-ibeos` passes.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
please abort

**Comment by mmusich** [comment]
please test

**Comment by emanueleusai** [comment]
+1

File: DQMOffline/Alignment/src/TkAlCaRecoMonitor.cc [hunk 1]
```diff
@@ -222,15 +224,14 @@ void TkAlCaRecoMonitor::analyze(const edm::Event &iEvent, const edm::EventSetup
...
     double dR = 0;
     if (runsOnReco_) {
       double minJetDeltaR = 10;  // some number > 2pi
-      for (reco::CaloJetCollection::const_iterator itJet = jets->begin(); itJet != jets->end(); ++itJet) {
-        jetPt_->Fill((*itJet).pt());
-        dR = deltaR((*track), (*itJet));
-        if ((*itJet).pt() > maxJetPt_ && dR < minJetDeltaR)
+      for (const auto &itJet : *jets) {
+        jetPt_->Fill(itJet.pt());
+        dR = deltaR(track, itJet);
```
Comment (perrotta) [inline DQMOffline/Alignment/src/TkAlCaRecoMonitor.cc:233]: Minor point: `deltaR2` is computationally less expensive

File: DQMOffline/Alignment/src/TkAlCaRecoMonitor.cc [hunk 2]
```diff
@@ -240,32 +241,28 @@ void TkAlCaRecoMonitor::analyze(const edm::Event &iEvent, const edm::EventSetup
     }
 
     double minTrackDeltaR = 10;  // some number > 2pi
-    for (reco::TrackCollection::const_iterator track2 = (*trackCollection).begin(); track2 < (*trackCollection).end();
-         ++track2) {
-      dR = deltaR((*track), (*track2));
+    for (const auto &track2 : *trackCollection) {
+      dR = deltaR(track, track2);
```
Comment (perrotta) [inline DQMOffline/Alignment/src/TkAlCaRecoMonitor.cc:245]: Minor point: `deltaR2` is computationally less expensive

File: DQMOffline/Alignment/interface/TkAlCaRecoMonitor.h [hunk 3]
```diff
@@ -36,20 +36,33 @@ class TkAlCaRecoMonitor : public DQMEDAnalyzer {
   explicit TkAlCaRecoMonitor(const edm::ParameterSet &);
   ~TkAlCaRecoMonitor() override = default;
 
+  static void fillDescriptions(edm::ConfigurationDescriptions &);
+
   void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
   void analyze(const edm::Event &, const edm::EventSetup &) override;
 
 private:
+  static constexpr const double kMuonMass_ = 0.10565836;
```
Comment (perrotta) [inline DQMOffline/Alignment/interface/TkAlCaRecoMonitor.h:45]: Ultra minor point: it is funny, there are slightly different values of the muon mass scattered in our code. The PDG value is 0.1056583755, tinily larger than what written here, Lacking a centrally accessible value of it in CMSSW, one could stick on the PDG value

**Review by perrotta — COMMENTED** [review]
These minor points could be left for a follow up update, if deemed worth to

**Comment by perrotta** [comment]
+1

File: DQMOffline/Alignment/src/TkAlCaRecoMonitor.cc (hunk 1 above)
Comment (mmusich) [reply DQMOffline/Alignment/src/TkAlCaRecoMonitor.cc:233]: but then isn't taking the root moving the expense elsewhere (notice we are interested in dR, not dR2). Or are you suggesting to plot directly dR2?

File: DQMOffline/Alignment/interface/TkAlCaRecoMonitor.h (hunk 3 above)
Comment (mmusich) [reply DQMOffline/Alignment/interface/TkAlCaRecoMonitor.h:45]: I would be really glad if we actually had to care about the sub per million differences in the muon mass discrepancy in the di-muon resonance mass that we are monitoring. We have much worse problems than this :D

File: DQMOffline/Alignment/src/TkAlCaRecoMonitor.cc (hunk 1 above)
Comment (perrotta) [reply DQMOffline/Alignment/src/TkAlCaRecoMonitor.cc:233]: As I said, it is a minor point. But yes, computing the square root only once before filling the histograms is less expensive than having to compute it at every jet or track

Comment (mmusich) [reply DQMOffline/Alignment/src/TkAlCaRecoMonitor.cc:233]: sorry, I admit I misunderstood what you proposed. Here's the requested improvement: https://github.com/cms-sw/cmssw/pull/40887.

File: DQMOffline/Alignment/interface/TkAlCaRecoMonitor.h (hunk 3 above)
Comment (mmusich) [reply DQMOffline/Alignment/interface/TkAlCaRecoMonitor.h:45]: In any case done at https://github.com/cms-sw/cmssw/pull/40887.

**Pull request merged**